infrastructure/constructs/lambda_config.py

```diff
@@ -1,6 +1,7 @@
-from aws_cdk import Duration, aws_lambda
+from aws_cdk import Duration, aws_lambda, aws_logs
 
 PYTHON_RUNTIME = aws_lambda.Runtime.PYTHON_3_9
+LOG_RETENTION = aws_logs.RetentionDays.THREE_MONTHS
```

Comment:
Three months seems a bit short - we've had more than that between releases, and in my experience it's often useful to compare the logs of an earlier release with the current release. In our case, I'd guess the stable system will eventually get to a point where dependency releases happen every few days semi-automatically, and actual feature/bug releases happen every few months (similar to the current linz-data-lake and bde-processor-deployment repos). In that case, having logs back a year or even more might be useful. What do you think, @billgeo?

Reply:
I am ok changing the length of time if we can come to an agreement (anything other than keeping the logs forever) 😃

Reply:
Another thing to consider is that we have exceeded 10,000 log groups in CI. This makes searching the most recent logs in Cloudwatch difficult (I was under the impression that logging wasn't even working). Keeping older logs could be useful, but I would much rather have easy access to recent logs. Maybe there is a way to archive older logs >3 months??? (I will investigate)...

Reply:
We need to consider the difference between logs and log groups. I don't expect there's a way to delete log *groups* automatically after a time. We should make sure we're only logging to a handful of groups in production, and then just delete all the others. Then our log retention policy could be much more lenient. If I understand correctly it should be possible to retain production log groups without giving them a static name, since most modifications to our stack shouldn't necessitate creating new log groups.

Reply:
Sure. I would say six months would do the trick if we want to keep them between releases.

Reply:
What about setting log retention to 1 year for prod and 3 months for non-prod?

Reply:
Sounds fine to me.